File: leetcode/java/0190_reverse_bits.java
public class Solution {
    // you need treat n as an unsigned value
    public int reverseBits(int n) {
        int rev = 0;

        for (int i = 0; i < 32; i++) {
            rev = rev | (n & 1);
            n >>= 1;

            if (i != 31) {
                rev <<= 1;
            }
        }
        
        return rev;
    }
}

File: leetcode/cpp/0190_reverse_bits.cpp
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        uint32_t ret = 0;

        for (int i = 0; i < 32; i++) {
            uint32_t leastSig = n & 1;
            n = n >> 1;
            ret += leastSig;

            if (i != 31) {
                ret = ret << 1;
            }
        }

        return ret;
    }
};